File: mobile/lib/models/reading_history.dart
import 'comic.dart';
import 'episode.dart';

/// Riwayat baca user (Lanjutkan Baca).
class ReadingHistory {
  final int id;
  final int comicId;
  final int episodeId;
  final int lastPage;
  final double progress;
  final bool isCompleted;
  final String updatedAt;
  final Comic? comic;
  final Episode? episode;

  const ReadingHistory({
    required this.id,
    required this.comicId,
    required this.episodeId,
    required this.lastPage,
    required this.progress,
    required this.isCompleted,
    required this.updatedAt,
    this.comic,
    this.episode,
  });

  factory ReadingHistory.fromJson(Map<String, dynamic> json) {
    return ReadingHistory(
      id: json['id'] as int,
      comicId: json['comic_id'] as int,
      episodeId: json['episode_id'] as int,
      lastPage: (json['last_page'] as num?)?.toInt() ?? 1,
      progress: (json['progress'] as num?)?.toDouble() ?? 0,
      isCompleted: json['is_completed'] as bool? ?? false,
      updatedAt: json['updated_at'] as String? ?? '',
      comic: json['comic'] != null ? Comic.fromJson(json['comic'] as Map<String, dynamic>) : null,
      episode: json['episode'] != null ? Episode.fromJson(json['episode'] as Map<String, dynamic>) : null,
    );
  }
}

File: backend/deploy/app/Models/ReadingHistory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ReadingHistory extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'comic_id',
        'episode_id',
        'last_page',
        'progress',
        'is_completed',
    ];

    protected $casts = [
        'is_completed' => 'boolean',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function comic(): BelongsTo
    {
        return $this->belongsTo(Comic::class);
    }

    public function episode(): BelongsTo
    {
        return $this->belongsTo(Episode::class);
    }
}
